File: backend/core-api/src/main/java/ai/govbiz/core/dto/sample/SampleItemPreparationResponse.java
package ai.govbiz.core.dto.sample;

import ai.govbiz.core.domain.sample.ProcessingStatus;
import ai.govbiz.core.domain.sample.SampleCategory;
import ai.govbiz.core.domain.sample.SampleItemPhase;

/** 예제 준비 API가 소유하는 공개 응답 DTO입니다. */
public record SampleItemPreparationResponse(
        SampleItemPhase phase,
        Item item,
        Processing processing
) {

    public record Item(String name, SampleCategory category, String note) {
    }

    public record Processing(ProcessingStatus status) {
    }
}

File: backend/core-api/src/main/kotlin/ai/govbiz/core/_sampleitem/controller/SampleItemPreparationResponse.kt
package ai.govbiz.core._sampleitem.controller

import ai.govbiz.core._sampleitem.domain.ProcessingStatus
import ai.govbiz.core._sampleitem.domain.SampleCategory
import ai.govbiz.core._sampleitem.domain.SampleItemPhase

/** 예제 준비 API가 소유하는 공개 응답 DTO입니다. */
data class SampleItemPreparationResponse(
    val phase: SampleItemPhase,
    val item: Item,
    val processing: Processing,
) {
    data class Item(
        val name: String,
        val category: SampleCategory?,
        val note: String?,
    )

    data class Processing(
        val status: ProcessingStatus,
    )
}
